backend/test/todos.test.ts

```diff
@@ -0,0 +1,287 @@
+import { gql } from 'apollo-server';
+import { createTestClient } from 'apollo-server-testing';
+import { GraphQLResponse } from 'apollo-server-types';
+import uuid = require('uuid');
+import { v4String } from 'uuid/interfaces';
+
+import { makeServer } from '../src/apollo';
+import { DEFAULT_TODOS, DEFAULT_USERS } from '../src/data';
+
+const testClient = createTestClient(
+  makeServer({
+    users: [...DEFAULT_USERS],
+    todos: [...DEFAULT_TODOS],
+  }),
+);
+
+describe('mutate', () => {
+  describe('createTodo', () => {
+    let response: GraphQLResponse;
+
+    const genMethodBackup = uuid.v4;
+    beforeAll(async () => {
+      (uuid.v4 as v4String) = () => {
+        return 'uuid';
+      };
+
+      const mutation = gql`
+        mutation {
+          createTodo(description: "test") {
+            id
+            description
+            isDone
+            assignedUser {
+              name
+            }
+          }
+        }
+      `;
+
+      response = await testClient.mutate({ mutation });
+      uuid.v4 = genMethodBackup;
+    });
+
+    it('adds a new todo', () => {
+      const data = (response.data && response.data.createTodo) || undefined;
+      expect(data).toMatchObject({
+        id: 'uuid',
+        description: 'test',
+        isDone: false,
+        assignedUser: null,
+      });
+    });
+
+    it('generates a random ID', () => {
+      const id = (response.data && response.data.createTodo && response.data.createTodo.id) || undefined;
+      expect(id).toBe('uuid');
+    });
+
+    it('initializes todo as not done', () => {
+      const isDone =
```

Comment:
```suggestion
      const { isDone } = response.data.createTodo
```

![complicated](https://media.giphy.com/media/SiKqz6IPADLF6IvONF/giphy.gif)

The ternary operator does not help at all but swallows a good information which we would otherwise see in the error message of the failing test.